File: ios-app/IzmirEshot/Models/Bus.swift
import Foundation

struct BusDto: Codable {
    let busId: Int
    let lineNo: Int
    let lineName: String
    let remainingStops: Int
    let latitudeRaw: String?
    let longitudeRaw: String?
    
    enum CodingKeys: String, CodingKey {
        case busId = "OtobusId"
        case lineNo = "HatNumarasi"
        case lineName = "HatAdi"
        case remainingStops = "KalanDurakSayisi"
        case latitudeRaw = "KoorX" // Mapped X to Latitude per analysis
        case longitudeRaw = "KoorY"
    }
    
    var latitude: Double? {
        latitudeRaw?.replacingOccurrences(of: ",", with: ".").toDouble()
    }
    
    var longitude: Double? {
        longitudeRaw?.replacingOccurrences(of: ",", with: ".").toDouble()
    }
}

extension String {
    func toDouble() -> Double? {
        return Double(self)
    }
}

File: android-app/app/src/main/java/com/eshot/app/data/model/Bus.kt
package com.eshot.app.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * API Response DTO for Approaching Buses
 * "KoorX": "38,403..." -> Latitude
 * "KoorY": "27,136..." -> Longitude
 */
@Serializable
data class BusDto(
    @SerialName("OtobusId") val busId: Int,
    @SerialName("HatNumarasi") val lineNo: Int,
    @SerialName("HatAdi") val lineName: String,
    @SerialName("KoorX") val latitudeRaw: String?, // String because of comma
    @SerialName("KoorY") val longitudeRaw: String?,
    @SerialName("KalanDurakSayisi") val remainingStops: Int,
    @SerialName("HattinYonu") val direction: Int
) {
    val latitude: Double?
        get() = latitudeRaw?.replace(",", ".")?.toDoubleOrNull()
        
    val longitude: Double?
        get() = longitudeRaw?.replace(",", ".")?.toDoubleOrNull()
}
